### ISSUE TITLE
Itinerary : Completing the gaps between appointments

### ISSUE THREAD

**Issue by GrosR** [clowdfish/ttg-frontend#85]
If you enter several appointments in a row, there are two possible scenarios: 
1) The appointments are not in sequence, so there might be one on the 1st of July, the other on the 7th of July, one on the 26th of July. 
In this case, it makes sense that we assume that all appointments might start at the same departure location. **However, we don´t know that.** 
That´s why we need the possiblity to change the departure location for any appointment. Damians Screens showed this from the beginning on, but due to the change in the left pane of the itinierary selection and the workflow, which does not allow direct click or editing of the appointments in the left pane, this feature is lost - and must be added again. 

2) An even more likely scenario is: The user adds appointments that are linked together as he manages a business trip to several appointments in a few days, like e.g. 
Starting destination: Basel
Appointment 1: 7th July, 9 - 11 AM, Zürich
Appointment 2: 7th July, 2-3 PM, Lucerne
Appointment 3: 8th July, 8 -10 AM, Geneva
Appointment 4: 8th July, 3 -6 PM, Bern

In this case, the starting destination for the 1st appointment is Basel, but for the 2nd, it is Zürich, for the 3rd it will be Lucerne, for the 4th it will be Geneva and for the return trip to Basel, the starting point will be Bern, as the last appointment ended here. 

On top, there are two additional things to tale into account: 
- between Lucerne and Geneva, the guy might want to go to a hotel ... or might want to return home (Basel). 
- After the last appointment, the user might either want to stay in Bern for an overnight.. or return home ..or to another place. 

**We must therefore allow the user to make his choice on the go in the itinerary screen.**

**Comment by damianmartone** [comment]
There's been a redesign in the itinerary screen for these cases, in the /questions.html

![screen shot 2016-07-29 at 11 16 53](https://cloud.githubusercontent.com/assets/8942032/17243614/054a384e-557e-11e6-846d-82a2da86b00f.png)

For the first case Rudi mentions:
- We change the logic in the sidebar, compressed items a bit, since we're going to have there not only appointments but other steps in the completion of the itinerary
- Besides appointments and return flights, we have alerts or questions that are placed between two items that have no correlativity. So if we detect that theres more than 24 hours between two appointments we can ask: Do you want to book a flight back and make that your new origin for the next trip? Do you wanna stay in a hotel those extra nights? do you want to add another appointment here where you have nothing? Do you want us to ignore this?
- Instead of the "change" button we have in each appointment, question items will have an ADD which basically allows to add an appointment between those dates, which will require to refresh the itinerary to include those.
- Basically no itinerary can be finished with questions on, the user has to decide to do something (or do nothing) through those days.

**Comment by GrosR** [comment]
Hi Damian, 
I like your suggestion and design. I still think that we need the option to 
- fly back to another destination
- select a kind of "skip this"
Do you agree?

**Comment by damianmartone** [comment]
Yes, i'd suggest a "SKIP" button next to the sidebar's title (itinerary selection) for the first thing. And the fly back to another destination will be done in the editing of the appointment (where you can see where you depart and change that)

**Comment by GrosR** [comment]
don´t you think that the "Skip"-button should be in the context of the current action? In this step, you would want to simply skip the question as you don´t want to choose any of the options - and you still want to proceed with the next appointment.

**Comment by damianmartone** [comment]
ah i thought you meant on skip the whole itinerary picking
yeah i havent added the action but it would just be another card saying 'do nothing / ignore'

**Comment by GrosR** [comment]
makes sense.
Skipping the whole itinerary picking is more like "accept all default and proceed". This could still make sense to have this option, but I doubt that we need that. I assume it`s mot very likely that users will just accept all defaults. And if it just one appointment, the users has the "next"-button

**Comment by davidtran** [comment]
So we have 4 options, right ? 
- Return to departure location
- Hotel
- Add appointment
- Skip
What happen when user click on Add appointment ?

**Comment by davidtran** [comment]
I think when user select return to home, we have to drop all future appointments as well, also we have to remove the return trip.

**Comment by GrosR** [comment]
If you click On 'Add an appointment', the user shall get an appointment card like the one on the search screen.
We must check with Damian, how this shall be displayed exactly , but I think it makes sense to display an appointment card and replace the questions. 
When the user saves the appointment, it will be added to the left bar and it will be displayed like any other appointment.

**Comment by damianmartone** [comment]
Ok the changes have been pushed to the templates repo. @davidtran @GrosR 

In first place, the user can change an existing appointment ONLY inthe itinerary selection (not in the travel options, since would all have to be recalculated and makes no sense).
To do so, once the appointment he wants is selected he can click on CHANGE in the sidebar item and get a screen like this

![screen shot 2016-08-01 at 17 09 55](https://cloud.githubusercontent.com/assets/8942032/17298638/0e5026da-580b-11e6-9f5e-129e9d2d0ff1.png)

@davidtran I just used a fake overlay but you can use any lightbox plugin you feel comfortable with, this was faster for me, and the styles are the same as the card at home except that I added a departure icon and an options icon. (intinerary-appointments.html file)

Here the user can modify both his options and the default between-appointments

Also the user can add a new appointment with the new Add Appointment button
![screen shot 2016-08-01 at 16 34 37](https://cloud.githubusercontent.com/assets/8942032/17298673/3e8fdd36-580b-11e6-99b5-ec9c12d71262.png)

Another minor change that I forgot in previous screens was that the title of each screen (h3 in the big part of the screen) has to be the name of the appointment and before that have the date as well this time with the day of the week too

**Comment by GrosR** [comment]
Hi Damian, 
wouldn´t it make more sense to use the same process that is already in place for the other appointments?
In Case a user clicks "Change" on an appointment in the itinerary section, shouldn´t we just start over for this appointment? 
I would understand the overlay for the use case "Add New Appointment".

Might be you have something in mind that I am missing,,,

**Comment by GrosR** [comment]
I just checked the workflow... you are right.
Still...I someone adds an appointment.. should he really enter what to do in between the appointments?
Or should he be guided through the process like with the other appointments?

**Comment by damianmartone** [comment]
he won't see this unless he decides to edit the appointment. so by default we will ask him like on the questions, and if he does edit the card, since the itinerary is gonna change, he might as well tell us in advanced so we dont ask again

**Comment by GrosR** [comment]
ok, understand, thanks

**Comment by davidtran** [comment]
Because user can change the date and time of the appointment, I think we need to sort appointments again after you edit/add appointment, right ?

**Comment by davidtran** [comment]
Should we delete the "No appointment" if user choose "Skip" ?

**Comment by GrosR** [comment]
Yes, resort the appointments, if date or time are changed.

In means of 'no appointment' it depends. This card will always be relevant if there is more than 24 hours in between two appointments.
So if you add an appointment or return flight and there is still a gap >24 hours without appointments , you should keep the now appointments card.

**Comment by damianmartone** [comment]
We delete the no appointment once the user has decided to skip it, since its no longer relevant, it was just so the user makes a decision

**Comment by GrosR** [comment]
ok, makes sense
